DevSkim failing both locally and on GH Actions

On my local environment, running DevSkim, this is the error I get:

```sh
Results of devskim linter (version 1.0.22)
See documentation on https://megalinter.io/7.4.0/descriptors/repository_devskim/
-----------------------------------------------

❌ [ERROR] for workspace /tmp/lint
Linter raw log:
git version 2.40.1
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
fatal: this operation must be run in a work tree
{"$schema":"https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.6.json","version":"2.1.0","runs":[{"tool":{"driver":{"name":"devskim","fullName":"Microsoft DevSkim Command Line Interface","version":"1.0.22+c8c65ea813","informationUri":"https://github.com/microsoft/DevSkim/","rules":[{"id":"DS176209","name":"SuspiciousComment","fullDescription":{"text":"Suspicious comment: A \"TODO\" or similar was left in source code, possibly indicating incomplete functionality"},"help":{"text":"A \"TODO\" or similar was left in source code, possibly indicating incomplete functionality","markdown":"Visit [https://github.com/Microsoft/DevSkim/blob/main/guidance/DS176209.md](https://github.com/Microsoft/DevSkim/blob/main/guidance/DS176209.md) for guidance on this issue."},"shortDescription":{"text":"A \"TODO\" or similar was left in source code, possibly indicating incomplete functionality"},"defaultConfiguration":{"level":"note"},"helpUri":"https://github.com/Microsoft/DevSkim/blob/main/guidance/DS176209.md","properties":{"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}},{"id":"DS126858","name":"WeakbrokenHashAlgorithm","fullDescription":{"text":"Weak/Broken Hash Algorithm: A weak or broken hash algorithm was detected."},"help":{"text":"Consider switching to use SHA-256 or SHA-512 instead.","markdown":"Visit [https://github.com/Microsoft/DevSkim/blob/main/guidance/DS126858.md](https://github.com/Microsoft/DevSkim/blob/main/guidance/DS126858.md) for guidance on this issue."},"shortDescription":{"text":"A weak or broken hash algorithm was detected."},"defaultConfiguration":{"level":"error"},"helpUri":"https://github.com/Microsoft/DevSkim/blob/main/guidance/DS126858.md","properties":{"DevSkimSeverity":"Critical","DevSkimConfidence":"Unspecified"}}]}},"results":[{"ruleId":"DS126858","level":"error","message":{"text":"Weak/Broken Hash Algorithm"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"region":{"startLine":70,"startColumn":18,"endLine":70,"endColumn":22,"charOffset":1998,"charLength":4,"snippet":{"text":"SHA1","rendered":{"text":"SHA1","markdown":"`SHA1`"}},"sourceLanguage":"yaml"}}}],"fixes":[{"description":{"text":"A weak or broken hash algorithm was detected."},"artifactChanges":[{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"replacements":[{"deletedRegion":{"charOffset":1998,"charLength":4},"insertedContent":{"text":"SHA256"}}]}]},{"description":{"text":"A weak or broken hash algorithm was detected."},"artifactChanges":[{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"replacements":[{"deletedRegion":{"charOffset":1998,"charLength":4},"insertedContent":{"text":"SHA512"}}]}]}],"properties":{"tags":["Cryptography.BannedHashAlgorithm"],"DevSkimSeverity":"Critical","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS126858","level":"error","message":{"text":"Weak/Broken Hash Algorithm"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"region":{"startLine":70,"startColumn":8,"endLine":70,"endColumn":13,"charOffset":1988,"charLength":5,"snippet":{"text":"SHA-1","rendered":{"text":"SHA-1","markdown":"`SHA-1`"}},"sourceLanguage":"yaml"}}}],"fixes":[{"description":{"text":"A weak or broken hash algorithm was detected."},"artifactChanges":[{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"replacements":[{"deletedRegion":{"charOffset":1988,"charLength":5},"insertedContent":{"text":"SHA256"}}]}]},{"description":{"text":"A weak or broken hash algorithm was detected."},"artifactChanges":[{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"replacements":[{"deletedRegion":{"charOffset":1988,"charLength":5},"insertedContent":{"text":"SHA512"}}]}]}],"properties":{"tags":["Cryptography.BannedHashAlgorithm"],"DevSkimSeverity":"Critical","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS126858","level":"error","message":{"text":"Weak/Broken Hash Algorithm"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"region":{"startLine":70,"startColumn":2,"endLine":70,"endColumn":6,"charOffset":1982,"charLength":4,"snippet":{"text":"SHA1","rendered":{"text":"SHA1","markdown":"`SHA1`"}},"sourceLanguage":"yaml"}}}],"fixes":[{"description":{"text":"A weak or broken hash algorithm was detected."},"artifactChanges":[{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"replacements":[{"deletedRegion":{"charOffset":1982,"charLength":4},"insertedContent":{"text":"SHA256"}}]}]},{"description":{"text":"A weak or broken hash algorithm was detected."},"artifactChanges":[{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"replacements":[{"deletedRegion":{"charOffset":1982,"charLength":4},"insertedContent":{"text":"SHA512"}}]}]}],"properties":{"tags":["Cryptography.BannedHashAlgorithm"],"DevSkimSeverity":"Critical","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS176209","level":"note","message":{"text":"Suspicious comment"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".vscode/settings.json"},"region":{"startLine":130,"startColumn":36,"endLine":130,"endColumn":40,"charOffset":5145,"charLength":4,"snippet":{"text":"TODO","rendered":{"text":"TODO","markdown":"`TODO`"}},"sourceLanguage":"json"}}}],"properties":{"tags":["Hygiene.Comment.Suspicious"],"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS176209","level":"note","message":{"text":"Suspicious comment"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".vscode/settings.json"},"region":{"startLine":130,"startColumn":5,"endLine":130,"endColumn":9,"charOffset":5114,"charLength":4,"snippet":{"text":"TODO","rendered":{"text":"TODO","markdown":"`TODO`"}},"sourceLanguage":"json"}}}],"properties":{"tags":["Hygiene.Comment.Suspicious"],"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS176209","level":"note","message":{"text":"Suspicious comment"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".vscode/settings.json"},"region":{"startLine":129,"startColumn":28,"endLine":129,"endColumn":32,"charOffset":5092,"charLength":4,"snippet":{"text":"TODO","rendered":{"text":"TODO","markdown":"`TODO`"}},"sourceLanguage":"json"}}}],"properties":{"tags":["Hygiene.Comment.Suspicious"],"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS176209","level":"note","message":{"text":"Suspicious comment"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".vscode/settings.json"},"region":{"startLine":127,"startColumn":69,"endLine":127,"endColumn":73,"charOffset":5012,"charLength":4,"snippet":{"text":"TODO","rendered":{"text":"TODO","markdown":"`TODO`"}},"sourceLanguage":"json"}}}],"properties":{"tags":["Hygiene.Comment.Suspicious"],"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS176209","level":"note","message":{"text":"Suspicious comment"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".config/linters/vale/styles/Google/Acronyms.yml"},"region":{"startLine":56,"startColumn":4,"endLine":56,"endColumn":8,"charOffset":751,"charLength":4,"snippet":{"text":"TODO","rendered":{"text":"TODO","markdown":"`TODO`"}},"sourceLanguage":"yaml"}}}],"properties":{"tags":["Hygiene.Comment.Suspicious"],"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS176209","level":"note","message":{"text":"Suspicious comment"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".vscode/settings.json"},"region":{"startLine":127,"startColumn":61,"endLine":127,"endColumn":65,"charOffset":5004,"charLength":4,"snippet":{"text":"TODO","rendered":{"text":"TODO","markdown":"`TODO`"}},"sourceLanguage":"json"}}}],"properties":{"tags":["Hygiene.Comment.Suspicious"],"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}}],"columnKind":"utf16CodeUnits"}]}
```

So it seems that DevSkim via MegaLinter is outputting non-human-readable JSON results into both the error logs and the console... making it a pain to figure out what's going wrong. There also seems to be some major errors going on somewhere along the line with `this operation must be run in a work tree`, but I have no idea what that is about.

Interestingly, on GitHub Actions, the behaviors and failures are different:

```
❌ Linted [REPOSITORY] files with [devskim]: Found 9 error(s) - (1.59s) (expand for details)
  - Using [devskim v1.0.18] https://megalinter.io/7.3.0/descriptors/repository_devskim
  - MegaLinter key: [REPOSITORY_DEVSKIM]
  - Rules config: identified by [devskim]
  --Error detail:
  git version 2.38.5
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  fatal: detected dubious ownership in repository at '/github/workspace/.git'
  To add an exception for this directory, call:
  
  	git config --global --add safe.directory /github/workspace/.git
  {"$schema":"https://www.schemastore.org/schemas/json/sarif-2.1.0.json","version":"2.1.0","runs":[{"tool":{"driver":{"name":"devskim","fullName":"Microsoft DevSkim Command Line Interface","version":"1.0.18+fba56c26b8","informationUri":"https://github.com/microsoft/DevSkim/","rules":[{"id":"DS176[209](https://github.com/andrewvaughan/template-core/actions/runs/6551575924/job/17793054905#step:4:219)","name":"SuspiciousComment","fullDescription":{"text":"Suspicious comment: A \"TODO\" or similar was left in source code, possibly indicating incomplete functionality"},"help":{"text":"A \"TODO\" or similar was left in source code, possibly indicating incomplete functionality","markdown":"Visit [https://github.com/Microsoft/DevSkim/blob/main/guidance/DS176209.md](https://github.com/Microsoft/DevSkim/blob/main/guidance/DS176209.md) for guidance on this issue."},"shortDescription":{"text":"A \"TODO\" or similar was left in source code, possibly indicating incomplete functionality"},"defaultConfiguration":{"level":"note"},"helpUri":"https://github.com/Microsoft/DevSkim/blob/main/guidance/DS176209.md","properties":{"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}},{"id":"DS126858","name":"WeakbrokenHashAlgorithm","fullDescription":{"text":"Weak/Broken Hash Algorithm: A weak or broken hash algorithm was detected."},"help":{"text":"Consider switching to use SHA-256 or SHA-512 instead.","markdown":"Visit [https://github.com/Microsoft/DevSkim/blob/main/guidance/DS126858.md](https://github.com/Microsoft/DevSkim/blob/main/guidance/DS126858.md) for guidance on this issue."},"shortDescription":{"text":"A weak or broken hash algorithm was detected."},"defaultConfiguration":{"level":"error"},"helpUri":"https://github.com/Microsoft/DevSkim/blob/main/guidance/DS126858.md","properties":{"DevSkimSeverity":"Critical","DevSkimConfidence":"Unspecified"}}]}},"versionControlProvenance":[{"repositoryUri":"https://github.com/andrewvaughan/template-core","revisionId":"fdf41cdba058f4181bd2f2a24486ecf6eb1e0533","branch":"(no branch)"}],"results":[{"ruleId":"DS176209","level":"note","message":{"text":"Suspicious comment"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".config/linters/vale/styles/Google/Acronyms.yml"},"region":{"startLine":56,"startColumn":4,"endLine":56,"endColumn":8,"charOffset":751,"charLength":4,"snippet":{"text":"TODO","rendered":{"text":"TODO","markdown":"`TODO`"}},"sourceLanguage":"yaml"}}}],"properties":{"tags":["Hygiene.Comment.Suspicious"],"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS126858","level":"error","message":{"text":"Weak/Broken Hash Algorithm"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"region":{"startLine":70,"startColumn":18,"endLine":70,"endColumn":22,"charOffset":1998,"charLength":4,"snippet":{"text":"SHA1","rendered":{"text":"SHA1","markdown":"`SHA1`"}},"sourceLanguage":"yaml"}}}],"fixes":[{"description":{"text":"A weak or broken hash algorithm was detected."},"artifactChanges":[{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"replacements":[{"deletedRegion":{"charOffset":1998,"charLength":4},"insertedContent":{"text":"SHA256"}}]}]},{"description":{"text":"A weak or broken hash algorithm was detected."},"artifactChanges":[{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"replacements":[{"deletedRegion":{"charOffset":1998,"charLength":4},"insertedContent":{"text":"SHA512"}}]}]}],"properties":{"tags":["Cryptography.BannedHashAlgorithm"],"DevSkimSeverity":"Critical","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS126858","level":"error","message":{"text":"Weak/Broken Hash Algorithm"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"region":{"startLine":70,"startColumn":8,"endLine":70,"endColumn":13,"charOffset":1988,"charLength":5,"snippet":{"text":"SHA-1","rendered":{"text":"SHA-1","markdown":"`SHA-1`"}},"sourceLanguage":"yaml"}}}],"fixes":[{"description":{"text":"A weak or broken hash algorithm was detected."},"artifactChanges":[{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"replacements":[{"deletedRegion":{"charOffset":1988,"charLength":5},"insertedContent":{"text":"SHA256"}}]}]},{"description":{"text":"A weak or broken hash algorithm was detected."},"artifactChanges":[{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"replacements":[{"deletedRegion":{"charOffset":1988,"charLength":5},"insertedContent":{"text":"SHA512"}}]}]}],"properties":{"tags":["Cryptography.BannedHashAlgorithm"],"DevSkimSeverity":"Critical","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS126858","level":"error","message":{"text":"Weak/Broken Hash Algorithm"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"region":{"startLine":70,"startColumn":2,"endLine":70,"endColumn":6,"charOffset":1982,"charLength":4,"snippet":{"text":"SHA1","rendered":{"text":"SHA1","markdown":"`SHA1`"}},"sourceLanguage":"yaml"}}}],"fixes":[{"description":{"text":"A weak or broken hash algorithm was detected."},"artifactChanges":[{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"replacements":[{"deletedRegion":{"charOffset":1982,"charLength":4},"insertedContent":{"text":"SHA256"}}]}]},{"description":{"text":"A weak or broken hash algorithm was detected."},"artifactChanges":[{"artifactLocation":{"uri":".config/linters/vale/styles/Google/WordList.yml"},"replacements":[{"deletedRegion":{"charOffset":1982,"charLength":4},"insertedContent":{"text":"SHA512"}}]}]}],"properties":{"tags":["Cryptography.BannedHashAlgorithm"],"DevSkimSeverity":"Critical","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS176209","level":"note","message":{"text":"Suspicious comment"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".vscode/settings.json"},"region":{"startLine":130,"startColumn":36,"endLine":130,"endColumn":40,"charOffset":5145,"charLength":4,"snippet":{"text":"TODO","rendered":{"text":"TODO","markdown":"`TODO`"}},"sourceLanguage":"json"}}}],"properties":{"tags":["Hygiene.Comment.Suspicious"],"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS176209","level":"note","message":{"text":"Suspicious comment"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".vscode/settings.json"},"region":{"startLine":130,"startColumn":5,"endLine":130,"endColumn":9,"charOffset":5114,"charLength":4,"snippet":{"text":"TODO","rendered":{"text":"TODO","markdown":"`TODO`"}},"sourceLanguage":"json"}}}],"properties":{"tags":["Hygiene.Comment.Suspicious"],"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS176209","level":"note","message":{"text":"Suspicious comment"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".vscode/settings.json"},"region":{"startLine":129,"startColumn":28,"endLine":129,"endColumn":32,"charOffset":5092,"charLength":4,"snippet":{"text":"TODO","rendered":{"text":"TODO","markdown":"`TODO`"}},"sourceLanguage":"json"}}}],"properties":{"tags":["Hygiene.Comment.Suspicious"],"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS176209","level":"note","message":{"text":"Suspicious comment"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".vscode/settings.json"},"region":{"startLine":127,"startColumn":69,"endLine":127,"endColumn":73,"charOffset":5012,"charLength":4,"snippet":{"text":"TODO","rendered":{"text":"TODO","markdown":"`TODO`"}},"sourceLanguage":"json"}}}],"properties":{"tags":["Hygiene.Comment.Suspicious"],"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}},{"ruleId":"DS176209","level":"note","message":{"text":"Suspicious comment"},"locations":[{"physicalLocation":{"artifactLocation":{"uri":".vscode/settings.json"},"region":{"startLine":127,"startColumn":61,"endLine":127,"endColumn":65,"charOffset":5004,"charLength":4,"snippet":{"text":"TODO","rendered":{"text":"TODO","markdown":"`TODO`"}},"sourceLanguage":"json"}}}],"properties":{"tags":["Hygiene.Comment.Suspicious"],"DevSkimSeverity":"ManualReview","DevSkimConfidence":"Unspecified"}}],"columnKind":"utf16CodeUnits"}]}
```

I believe what is happening here is that there are _additional_ issues on DevSkim from CVE-2022–24765. [Details here](https://medium.com/@janloo/github-actions-detected-dubious-ownership-in-repository-at-github-workspace-how-to-fix-b9cc127d4c04).  I believe this can be avoided by simply ensuring this command is run on the environment when configuring git:

```sh
git config --global --add safe.directory /github/workspace
```

I will test this with `PRE_COMMANDS` and see if I have improvement.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,6 +19,7 @@ Note: Can be used with `oxsecurity/megalinter@beta` in your GitHub Action mega-l
 - Fixes
   - build.py: Remove exclusivity between pip, gem & cargo packages
   - Salesforce linters: Switch sfdx-cli to @salesforce/cli
+  - Added default `.devskim.json` to mitigate errors introduced when no config exists
 
 - Doc
   - Display list of articles from newest to oldest

--- a/TEMPLATES/.devskim.json
+++ b/TEMPLATES/.devskim.json
@@ -1,0 +1,3 @@
+{
+  "Globs": ["**/.git/**", "**/megalinter-reports/**"]
+}

--- a/docs/descriptors/repository_devskim.md
+++ b/docs/descriptors/repository_devskim.md
@@ -7,16 +7,22 @@ description: How to use devskim (configure, ignore files, ignore errors, help & 
 # <a href="https://github.com/microsoft/DevSkim" target="blank" title="Visit linter Web Site"><img src="https://github.com/microsoft/DevSkim/raw/main/media/devskim_logo.svg" alt="devskim" height="100px" class="megalinter-logo"></a>devskim
 [![GitHub stars](https://img.shields.io/github/stars/microsoft/DevSkim?cacheSeconds=3600)](https://github.com/microsoft/DevSkim) ![sarif](https://shields.io/badge/-SARIF-orange) [![GitHub release (latest SemVer)](https://img.shields.io/github/v/release/microsoft/DevSkim?sort=semver)](https://github.com/microsoft/DevSkim/releases) [![GitHub last commit](https://img.shields.io/github/last-commit/microsoft/DevSkim)](https://github.com/microsoft/DevSkim/commits) [![GitHub commit activity](https://img.shields.io/github/commit-activity/y/microsoft/DevSkim)](https://github.com/microsoft/DevSkim/graphs/commit-activity/) [![GitHub contributors](https://img.shields.io/github/contributors/microsoft/DevSkim)](https://github.com/microsoft/DevSkim/graphs/contributors/)
 
-Use `--ignore-globs` to ignore files and/or folders
+Use the `Globs` configuration in a `.devskim.json` configuration file to ignore files and/or folders.
 
 Example:
-  `REPOSITORY_DEVSKIM_ARGUMENTS: ['--ignore-globs','**/megalinter-reports/**,**/.git/**,**/bin/**']`
+
+```json
+{
+  "Globs": ["**/.git/**", "**/megalinter-reports/**"]
+}
+```
 
 ## devskim documentation
 
 - Version in MegaLinter: **1.0.23**
 - Visit [Official Web Site](https://github.com/microsoft/DevSkim#readme){target=_blank}
 - See [How to configure devskim rules](https://github.com/microsoft/DevSkim/wiki/Analyze-Command){target=_blank}
+  - If custom `.devskim.json` config file isn't found, [.devskim.json](https://github.com/oxsecurity/megalinter/tree/main/TEMPLATES/.devskim.json){target=_blank} will be used
 - See [How to ignore files and directories with devskim](https://github.com/microsoft/DevSkim/wiki/Analyze-Command){target=_blank}
 
 [![DevSkim - GitHub](https://gh-card.dev/repos/microsoft/DevSkim.svg?fullname=)](https://github.com/microsoft/DevSkim){target=_blank}

--- a/megalinter/descriptors/repository.megalinter-descriptor.yml
+++ b/megalinter/descriptors/repository.megalinter-descriptor.yml
@@ -56,10 +56,18 @@ linters:
       - security
     ignore_for_flavor_suggestions: true
     linter_text: |
-      Use `--ignore-globs` to ignore files and/or folders
+      If you need to ignore folders,files or file extensions, use glob expressions `Glob` property of local `.devskim.json` file
 
       Example:
-        `REPOSITORY_DEVSKIM_ARGUMENTS: ['--ignore-globs','**/megalinter-reports/**,**/.git/**,**/bin/**']`
+
+      ```json
+      {
+        Glob: [
+          "**/.git/**",
+          "**/megalinter-reports/**"
+        ]
+      }
+      ```
     linter_url: https://github.com/microsoft/DevSkim
     linter_repo: https://github.com/microsoft/DevSkim
     linter_speed: 1

--- a/megalinter/descriptors/repository.megalinter-descriptor.yml
+++ b/megalinter/descriptors/repository.megalinter-descriptor.yml
@@ -75,6 +75,7 @@ linters:
     linter_rules_ignore_config_url: https://github.com/microsoft/DevSkim/wiki/Analyze-Command
     linter_image_url: https://github.com/microsoft/DevSkim/raw/main/media/devskim_logo.svg
     linter_icon_png_url: https://raw.githubusercontent.com/oxsecurity/megalinter/main/docs/assets/icons/linters/devskim.png
+    config_file_name: .devskim.json
     cli_lint_mode: project
     cli_lint_extra_args:
       - analyze

